tradefed/__init__.py

```diff
@@ -224,13 +236,18 @@ def _extract_cts_results(self, buf, testrun, suite_prefix):
             logger.debug("creating suite metadata: {}".format(atomic_test_suite_name))
             suite_metadata, _ = SuiteMetadata.objects.get_or_create(suite=atomic_test_suite_name, kind='suite')
             suite, _ = Suite.objects.get_or_create(slug=atomic_test_suite_name, project=testrun.build.project, defaults={"metadata": suite_metadata})
-            #logger.debug("Adding status with suite: {suite_prefix}/{module_name}".format(suite_prefix=suite_prefix, module_name=module_name))
-            #logger.debug("Creating subtasks for extracting results")
-            #task_list = task_list + [create_testcase_tests.s(ET.tostring(test_case, encoding="utf-8"), module_name, testrun.pk, suite.pk) for test_case in test_cases]
+            logger.debug("Adding status with suite: {suite_prefix}/{module_name}".format(suite_prefix=suite_prefix, module_name=module_name))
+            logger.debug("Creating subtasks for extracting results")
             for test_case in test_cases:
-                create_testcase_tests(ET.tostring(test_case, encoding="utf-8"), atomic_test_suite_name, testrun.pk, suite.pk)
-
-        #celery_chord(task_list)(update_build_status.s(testrun.pk))
+                plugin_scratch = PluginScratch.objects.create(
+                    build=testrun.build,
+                    storage=ET.tostring(test_case).decode('utf-8')
+                )
+                logger.debug("Created plugin scratch with ID: %s" % plugin_scratch.pk)
+                task = create_testcase_tests.s(plugin_scratch.pk, module_name, testrun.pk, suite.pk)
+                task_list.append(task)
+
+        celery_chord(task_list)(update_build_status.s(testrun.pk))
```

Comment:
One last thing I forgot to ask: what happens if a task from `task_list` fails for some reason? Would the other tasks be run as well?

Reply:
other tasks will be executed as they're already in the queue. What will not be executed is update_build_status. Tasks from the task_list should not fail because of this reason. They should always exit nicely even if they don't do what they are supposed to. There is also a way to execute a chained task on error, but I didn't explore this avenue yet.